Marketplace/azoozScripts/hookgun_azooz.js

```diff
@@ -0,0 +1,289 @@
+      return Object.keys(obj).length === 0 && JSON.stringify(obj) === JSON.stringify({});
+  }
+  function rayCollidable(origin, direction){
+    var ray = {origin: origin, direction: direction};
+    var trace = Entities.findRayIntersection(ray, true);
+    var blacklist = [];
+    var properties = Entities.getEntityProperties(trace.entityID , ["name", "collisionless", "visible"]);
+	//print("CMF: "+properties.name);
+    while(trace.intersects && (properties.collisionless  || properties.name!="HookSpot")) {
+      blacklist.push(trace.entityID);
+      trace = Entities.findRayIntersection(ray, true, [], blacklist);
+    }
+    return trace;
+  }
+  var getWeaponBulletInfo = function(weaponInstance) {
+
+      var properties = weaponInstance.properties;
+      if (emptyObjectCheck(properties)) {
+          return {
+              found: false
+          };
+      }
+
+      var frontVector = Quat.getFront(properties.rotation);
+      var frontOffset = Vec3.multiply(frontVector, GUN_TIP_FWD_OFFSET);
+      var bulletSpawnOffset = Vec3.multiply(frontVector, GUN_TIP_FWD_OFFSET + BULLET_SPAWN_OFFSET);
+      var tip = Vec3.multiply(frontVector, GUN_TIP_FWD_OFFSET + BULLET_SPAWN_OFFSET);
+      var upVector = Quat.getUp(properties.rotation);
+      var upOffset = Vec3.multiply(upVector, GUN_TIP_UP_OFFSET);
+
+      var forward = Quat.getFront(properties.rotation);
+      var gunTipPosition = Vec3.sum(properties.position, frontOffset);
+      var bulletSpawnPosition = Vec3.sum(properties.position, bulletSpawnOffset);
+
+      return {
+          found: true,
+          forward: forward,
+          tipSpawn: bulletSpawnPosition,
+          tip: gunTipPosition,
+          rotation: properties.rotation,
+          position: properties.position
+      };
+
+  }
+
+
+  var ActiveHand = {
+      0: {
+          secured: false
+      },
+      1: {
+          secured: false
+      }
+  };
+  var ActiveEquip = false;
+
+  //var debugLine = new Overlay({type: "line3d", position: Vec3.ZERO, color: {blue:Math.random()*255, red: Math.random()*255, green: Math.random()*255}}).addOverlay();
+
+  var HookShot = function (val) {
+    Entity.call(this, val);
+  };
+  HookShot.prototype = Object.create(Entity.prototype);
+  HookShot.prototype.ropeTo = function (fromPosition, toPosition) {
+      var rotation = Quat.multiply(Quat.lookAt(fromPosition, toPosition, Vec3.UP), {
+          w: 0.707,
+          z: 0,
+          x: 0.707,
+          y: 0
+      });
+      var distance = Vec3.distance(toPosition, fromPosition);
+      if (distance >= MAX_DISTANCE) {
+          return {distance: MAX_DISTANCE,
+            rotation: rotation};
+      }
+
+      return {
+          distance: distance,
+          rotation: rotation
+      }
+  };
+  HookShot.prototype.calculateThrustDirection = function(fromPosition, toPosition) {
+      return Vec3.normalize(Vec3.sum(toPosition, Vec3.multiply(fromPosition, -1)));
+  };
+  gun = new HookShot(function(id){

```

Comment:
this function is indented 2 instead of 4 spaces -- not sure why eslint didn't grab it